salt/orch/update.sls

```diff
@@ -134,6 +160,20 @@ update_modules:
     - arg:
       - update_in_progress
       - true
+    - require:
+      # wait until all the masters have been updated
+{%- for master_id in masters.keys() %}
+      - salt: {{ master_id }}-remove-update-grain
+{%- endfor %}
+
+# Perform any migrations necessary before shutting down services
+{{ worker_id }}-pre-shutdown:
+  salt.state:
+    - tgt: {{ worker_id }}
+    - sls:
+      - cni.update-pre-shutdown
+    - require:
+      - salt: {{ worker_id }}-set-update-grain
```

Comment:
I think doing this here is too late. Maybe we can run this migration before even restarting the master. I'm still taking a look at this, but looks like when we first update the master and reboot it with CNI the rest of the kubelets will try to re-register against it, and they will be provided their own `podCIDR` because they are lacking it.

We then reach this point and on most of the kubelets they already have a newly assigned `podCIDR` and we do nothing here (and of course the `podCIDR` won't match the network range before updating).